tailscale/data_source_user.go

```diff
@@ -78,32 +68,78 @@ func dataSourceUser() *schema.Resource {
 	return &schema.Resource{
 		Description: "The user data source describes a single user in a tailnet",
 		ReadContext: dataSourceUserRead,
-		Schema:      userSchema,
+		Schema: combinedSchemas(commonUserSchema, map[string]*schema.Schema{
+			"login_name": {
+				Type:         schema.TypeString,
+				Description:  "The emailish login name of the user.",
+				Optional:     true,
+				AtLeastOneOf: []string{"display_name", "login_name"},
```

Comment:
Should this be `ExactlyOneOf` instead of `AtLeastOneOf`, or are their cases where a user would want to be specifying both and only specifying one wouldn't be enough to identify a single user?

Reply:
Per discussion, I'm only allowing use of `login_name` now.